app/models/services/locators/doi_paper_locator.rb

```diff
@@ -6,29 +6,25 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
+    crossref = Crossref.new
+    if result = crossref.get_doi_metadata(self.locator_id)
+      data = result['message']
+      paper = self.import_data_to_paper(paper,data,'crossref')
+    end
+
     pubmed = Pubmed.new
     if uid = pubmed.find_uid_by_doi(self.locator_id)
       result = pubmed.get_uid_metadata(uid)
       data = result['result'][uid]
-      names = data['authors'].map {|a| a['name']}
-
-      existing_authors = Author.where(name: names)
-      existing_names = existing_authors.map(&:name)
-      new_authors = (names - existing_names).map {|a| Author.create name: a}
+      paper = self.import_data_to_paper(paper,data,'pubmed')
+    end
 
-      paper = Paper.create(
-        pubmed_id: data['uid'],
-        title: data['title'],
-        published_at: data['pubdate'],
-        authors: (existing_authors + new_authors),
-        abstract: pubmed.get_abstract(data['uid']),
-        doi: self.locator_id,
-        publication: data['source']
-      )
+    if paper.save
       return paper
     else
-      paper = Paper.new
-      paper.errors.add(:locator_id, "is invalid; no paper found for searched DOI: #{self.locator_id}")
+      paper.errors.add(:locator_id, "is invalid; no paper found for searched DOI: #{self.locator_id}") if paper.errors.empty?
```

Comment:
this is prob a short term fix.  there's a bigger issue going on now that we're pulling from multiple sources which is finding papers in credo based on multiple locators.  So the base locator only searches for a paper in our db based on the locator's type.  It will need to search based on doi, uid, or title to remove duplication.  

The reason I put that condition there was because it was throwing a "title is already taken" error if I tried to say submit a DOI for a paper that was already submitted by link and title and didn't have a DOI.  This would trigger the error message "no paper found with this locator id" along side of "Title has already been taken", but it's actually just the second one that was the issue. 

I think if I solve the bigger problem I won't need this conditional.